x-pack/plugins/ingest_manager/public/applications/ingest_manager/types/intra_app_route_state.ts

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { ApplicationStart } from 'kibana/public';
+import { Datasource } from '../../../../common/types/models';
+
+/**
+ * Supported routing state for the create datasource page routes
+ */
+export interface CreateDatasourceRouteState {
```

Comment:
Created this new types file under public to expose the interfaces supported for intra app state sharing. Currently only one for Create Datasources, but if other flows are added in the future, they can just be added here.